### DISCUSSION TITLE
Set variable with the first array item of a response

### DISCUSSION THREAD

**Discussion by quammy** [Huachao/vscode-restclient#1167] [Q&A] [answered]
How can I set a variable with value of a property of an object that is the first item of a response json array?
Here is a response sample:
```
[
  {
    "id": 4267358,
    "is_active": true,
    "name": "App1",
    "pcn_business_experience": null
  }
]
```

and this is my REST Client code:
```
@test1 = {{companies.response.body[0].id}}           ///////// this doesn't work

### get all companies
// @name companies
GET {{BaseURL}}/rest/v1.0/companies
Content-Type: application/json
```
Thanks!

**Comment by anmaso** [comment, marked as answer]
I've used this successfully (note the `$` after the body )
```
@test1 = {{request1.response.body.$.results[0].id}}
```

**Reply by quammy** [reply]
Perfect, thank you!
this one worked for me:
```@test1 = {{request1.response.body.$.[0].id}}```

**Comment by MominRaza** [comment]
Related update: https://github.com/Huachao/vscode-restclient/pull/1422

This discussion is addressed by the new `@set` directive, which lets you capture response values (`response.headers.*` / `response.body.*`) into shared runtime variables and reuse them across `.http` files.
